src/main/kotlin/com/personio/synthetics/builder/api/StepBuilder.kt

```diff
@@ -87,4 +92,24 @@ class StepBuilder(
             parsingOptions.add(options)
         }
     }
+
+    /**
+     * Sets the retry count and interval for the step
+     * @param retryCount The retry count for the step
+     * Allowed retry count is between 0 and 5
+     * @param retryInterval The retry interval for the step
+     * Allowed retry interval is between 0 and 5 seconds
+     */
+    fun retry(
+        retryCount: Long,
+        retryInterval: Duration,
+    ) {
+        require(retryCount in 0..5) {
+            "Step retry count should be between 0 and 5."
+        }
+        require(retryInterval in 0.seconds..5.seconds) {
+            "Step retry interval should be between 0 and 5000 milliseconds."
+        }
+        retryOptions = SyntheticsTestOptionsRetry().count(retryCount).interval(retryInterval.inWholeMilliseconds.toDouble())
```

Comment:
since the object is already initialized, you can use it `retryOptions.count(...)`